pkg/networkpolicy/multinetegressrule.go

```diff
@@ -54,6 +54,40 @@ func (builder *EgressRuleBuilder) WithPortAndProtocol(port uint16, protocol core
 	return builder
 }
 
+// WithProtocol appends new item with only protocol to Ports list.
+func (builder *EgressRuleBuilder) WithProtocol(protocol corev1.Protocol) *EgressRuleBuilder {
```

Comment:
since protocol is an enum, it may be good to validate it's one of the valid options, or at least not empty